src/control/server/storage/bdev/provider.go

```diff
@@ -152,7 +157,21 @@ func (p *Provider) IsVMDDisabled() bool {
 func (p *Provider) Scan(req ScanRequest) (*ScanResponse, error) {
 	if p.shouldForward(req) {
 		req.DisableVMD = p.IsVMDDisabled()
-		return p.fwd.Scan(req)
+
+		p.Lock()
+		defer p.Unlock()
+
+		if p.scanCache == nil || req.Rescan {
+			p.log.Debug("bdev provider rescan requested")
```

Comment:
Very minor but the message is a bit misleading in the case where the cache isn't populated. If you make any other changes before this lands, could say something like "populating bdev scan cache" instead

Reply:
will address in https://github.com/daos-stack/daos/pull/3423 if that's okay

Reply:
Works for me.

Reply:
done